packages/dds/tree/src/feature-libraries/flex-tree/flexTreeTypes.ts

```diff
@@ -173,16 +173,6 @@ export interface FlexTreeNode extends FlexTreeEntity {
 
 	boxedIterator(): IterableIterator<FlexTreeField>;
 
-	/**
-	 * The anchor node associated with this node
-	 *
-	 * @remarks
-	 * The ref count keeping this alive is owned by the FlexTreeNode:
-	 * if holding onto this anchor for longer than the FlexTreeNode might be alive,
-	 * a separate Anchor (and thus ref count) must be allocated to keep it alive.
-	 */
-	readonly anchorNode: AnchorNode;
```

Comment:
Only hydrated nodes have anchors: moving this to the new interface fixes that unhydrated nodes had to provide a dummy version which threw, and thus improves type safety. This is the largest improvement in this change: the rest is mostly fixes to make it still build.